src/dialects/postgres/query-generator.js

```diff
@@ -124,8 +124,39 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
     return `DROP TABLE IF EXISTS ${this.quoteTable(tableName)}${options.cascade ? ' CASCADE' : ''};`;
   }
 
-  showTablesQuery() {
-    return 'SELECT table_name FROM information_schema.tables WHERE table_schema = \'public\' AND table_type LIKE \'%TABLE\' AND table_name != \'spatial_ref_sys\';';
+  /**
+   *
+   * @param {*} _       Unused variable (database name)
+   * @param {object}    [options] Options
+   * @param {string}    [options.schema] Schema name to search for (defaults to 'public')
+   * @returns {string}  SQL statement to show all tables, including tableName and schema
+   */
+  showTablesQuery(_, options) {
+    const settings = {
+      schema: {
+        name: 'public',
+        op: '=',
+      },
+    };
+    settings.schema.name = options?.schema || settings.schema.name;
+
+    if (Array.isArray(settings.schema.name)) {
+      settings.schema.op = 'SIMILAR TO';
+      settings.schema.name = `(${settings.schema.name.join('|')})`;
+    }
+
+    if (['all', '*', '%'].includes(settings.schema.name?.toLowerCase())) {
+      settings.schema.op = 'LIKE';
+      settings.schema.name = '%';
+    }
```

Comment:
Good idea! Can we create this as a new issue?  My PR is starting to get longer :)

Reply:
This would reduce the size of your PR actually, since it touches a change you've introduced here

Reply:
It's important to note, where I use `Array.from().flat()`, the flattening is important.  I'd still be using `toArray().flat()`. I'm starting to question what this might buy.

```javascript
const arr = [
  [
    { users: ['fk1','fk2'] }, 
    { admins: ['fk1', 'fk2'] }
  ],
  { meta: 'stuff'}
];

toArray(arr)         // unchanged: [ [ { users: ['fk1','fk2'] }, { admins: ['fk1','fk2'] } ], { meta: 'stuff' } ]
Array.from(arr)    // unchanged: [ [ { users: ['fk1','fk2'] }, { admins: ['fk1','fk2'] } ], { meta: 'stuff' } ]

toArray(arr).flat()  // desired:  [ { users: [ 'fk1', 'fk2' ] }, { admins: [ 'fk1', 'fk2' ] }, { meta: 'stuff' } ]
Array.from(arr).flat()  // desired:  [ { users: [ 'fk1', 'fk2' ] }, { admins: [ 'fk1', 'fk2' ] }, { meta: 'stuff' } ]
```

I'm not sure what `toArray` buys from the native `Array.from` and I'm not sure what it would clean up.

Reply:
You shouldn't need to call flat: `toArray` does not wrap arrays in arrays.

The method in question here should definitely not accept a nested array input, so no flattening should be needed.